components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/KeyManagersApiServiceImpl.java

```diff
@@ -63,84 +77,262 @@ public Response keyManagersDiscoverPost(String url, String type, MessageContext
 
     public Response keyManagersGet(MessageContext messageContext) throws APIManagementException {
 
-        String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
+        String organization = RestApiUtil.getOrganization(messageContext);
         APIAdmin apiAdmin = new APIAdminImpl();
-        List<KeyManagerConfigurationDTO> keyManagerConfigurationsByTenant =
-                apiAdmin.getKeyManagerConfigurationsByTenant(tenantDomain);
+        List<KeyManagerConfigurationDTO> keyManagerConfigurationsByOrganization =
+                apiAdmin.getKeyManagerConfigurationsByOrganization(organization);
+        for (KeyManagerConfigurationDTO keyManagerConfigurationDTO: keyManagerConfigurationsByOrganization) {
+            if (StringUtils.equals(KeyManagerConfiguration.TokenType.EXCHANGED.toString(),
+                    keyManagerConfigurationDTO.getTokenType())) {
+                try {
+                    if (keyManagerConfigurationDTO.getExternalReferenceId() != null) {
+                        IdentityProvider identityProvider = IdentityProviderManager.getInstance()
+                                .getIdPByResourceId(keyManagerConfigurationDTO.getExternalReferenceId(),
+                                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, Boolean.FALSE);
+                        // Only two parameters that are common to IdentityProvider object and the KeyManagerInfoDTO (that
```

Comment:
let's not hardcode supertenant

Reply:
Sure. This will be covered via https://github.com/wso2/carbon-apimgt/pull/10600